components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -1006,15 +1006,12 @@ export const useSwap = () => {
         return 'insufficientAllowance'
       }
 
-      if (quoteErrorUnion?.zeroExError) {
-        if (quoteErrorUnion.zeroExError.isInsufficientLiquidity) {
-          return 'insufficientLiquidity'
-        }
-
-        if (quoteErrorUnion.zeroExError.isInsufficientAllowance) {
-          return 'insufficientAllowance'
-        }
+      // 0x specific validations
+      if (quoteUnion?.zeroExQuote?.liquidityAvailable === false) {
+        return 'insufficientLiquidity'
+      }
 
+      if (quoteErrorUnion?.zeroExError) {
```

Comment:
In this case, shouldn't  `quoteErrorUnion.zeroExError. isInsufficientLiquidity == true`?
checking both `QuoteUnion` and `QuoteErrorUnion` for errors seems odd to me. Also the `quoteErrorUnion.zeroExError.isInsufficientAllowance` case seems to be ignored.

Reply:
This was actually done because 0x API v2 now returns `isInsufficientLiquidity` as part of the quote instead of an error response. But now that I think about it, retaining the old interface is cleaner. You can find this change here: [review(nuo-xu): reuse ZeroExQuote and drop ZeroExQuoteInfo struct](https://github.com/brave/brave-core/pull/26231/commits/1bbee57c1c8414b6fb3fe409da02134df3741f56)

> Also the quoteErrorUnion.zeroExError.isInsufficientAllowance case seems to be ignored

Allowance checks are no longer returned as part of the error response from the API. If this changes in future, will add this back. Meanwhile, the desktop code already handles allowance checks just like LiFi and Squid.